doc/mir-json-schema.ts

```diff
@@ -0,0 +1,407 @@
+/// The MIR JSON format
```

Comment:
To accompany these `.ts` files, it would be helpful to have a `doc/README.md` file saying what these `*.ts` files, linking to the `simple-json-schema` repo as a resource for learning more. I think it would be helpful to link `doc/README.md` from the top-level README as well (probably in the [JSON schema section](https://github.com/GaloisInc/mir-json/blob/c9278101276acf40c191a28dd8bdbb9cbf08be86/README.md#json-schema)).